### PR TITLE
[UI Tests] Added a UI test for "Activity Log" dashboard card navigation.

### PR DESCRIPTION
**Pull request by pachlava**
### Description
Following its [iOS counterpart](https://github.com/wordpress-mobile/WordPress-iOS/pull/20796), this PR adds a UI test for the `Activity Log` dashboard card:

Screenshot 2023-06-07 at 20 15 53

What the test does:

```kotlin
@Test
fun e2eActivityLogCardNavigation() {
    MySitesPage()
        .scrollToActivityLogCard()
        .assertActivityLogCard()
        .assertActivityLogCardHasActivity("Enabled Jetpack Social")
        .assertActivityLogCardHasActivity("The Jetpack connection")
        .assertActivityLogCardHasActivity("This site is connected to")
        .tapActivity("The Jetpack connection")
        .assertEventScreenLoaded()
        .assertEventScreenHasActivity("The Jetpack connection is now complete. Welcome!")
}
```

Note that for Android, the test taps an individual activity and checks that `Event` screen is loaded. For iOS, the test tapped the card header (similar to tapping `View all activity`  link on Android), and checked that `Activity Log` screen is loaded. 

Firstly, I tried the same for Android. But I experienced weird flakiness when the test tapped `View all activity`. Every second time or so, the `Activity Log` screen failed to load the activities (but there were no unmatched requests warnings from WireMock). So, after trying to fix this with no success for too long, I decided to tap the individual activity and check the redirect to `Event` screen instead. This was reproducible only when using mocked API.

Since I added all the things (screen file, mocks) for `Activity Log` screen first, I ended up leaving them here, even though it's not used in this test. However, they might come handy later.

### To test
- All tests are green on CI, including the new `e2eActivityLogCardNavigation`

### Regression Notes
I'm not filling the regression notes since the PR adds a UI test, it does not change the app behaviour in any way.

### CODE REVIEW

File: WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/ActivityLogScreen.kt
Comment (pachlava) [inline WordPress/src/androidTest/java/org/wordpress/android/e2e/pages/ActivityLogScreen.kt:1]: This file ended up being unused, but I think it still makes sense to keep it.

File: libs/mocks/src/main/assets/mocks/mappings/wpcom/jetpack/wpcom-v2-sites-181977606-rewind-downloads.json
Comment (jostnes) [inline libs/mocks/src/main/assets/mocks/mappings/wpcom/jetpack/wpcom-v2-sites-181977606-rewind-downloads.json]: the site number on the `urlPath` no longer matches the site number on the file name after this change. looks like a typo to me or is this expected?

File: libs/mocks/src/main/assets/mocks/mappings/wpcom/jetpack/wpcom-v2-sites-181977606-rewind.json
Comment (jostnes) [inline libs/mocks/src/main/assets/mocks/mappings/wpcom/jetpack/wpcom-v2-sites-181977606-rewind.json]: same for this

**Review by jostnes — APPROVED** [review]
thanks for working on the Android equivalent so quickly! test works well locally and CI is 🟢 

i added a comment that i think should be updated before merging (looks like a typo) and i trust that you'll make the change before merging so approving to not block you later

File: libs/mocks/src/main/assets/mocks/mappings/wpcom/jetpack/wpcom-v2-sites-181977606-rewind-downloads.json
Comment (pachlava) [reply libs/mocks/src/main/assets/mocks/mappings/wpcom/jetpack/wpcom-v2-sites-181977606-rewind-downloads.json]: Thanks for catching this @jostnes 🆒 I renamed both files with e925b8dcd16aa8616e40d5ece6c29bda11cd5787.

**Comment by pachlava** [comment]
Thank you for the review @jostnes! I renamed the files and waiting for the 🟢 CI 🤞.

**Pull request merged**
